# Analzing Loan Data from Prosper
## by Aaron Remski

## Preliminary Wrangling

> This data set contains ~114k records containing information on various bank loans.

# Notes
Perceptually uniform palettes (seaborn) are: rocket, mako, flare, crest

* qualitative palettes, good for representing categorical data
* sequential palettes, good for representing numeric data
* diverging palettes, good for representing numeric data with a categorical boundary

In [2]:
# import all packages and set plots to be embedded inline
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sb

%matplotlib inline

> Exploratory visualizations & prorammatic investigations

> What kind of social variables correlate? Occupation & loan amount or Occupation & Borrow

In [3]:
loanDF = pd.read_csv("prosperLoanData.csv")

In [4]:
loanDF.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 113937 entries, 0 to 113936
Data columns (total 81 columns):
 #   Column                               Non-Null Count   Dtype  
---  ------                               --------------   -----  
 0   ListingKey                           113937 non-null  object 
 1   ListingNumber                        113937 non-null  int64  
 2   ListingCreationDate                  113937 non-null  object 
 3   CreditGrade                          28953 non-null   object 
 4   Term                                 113937 non-null  int64  
 5   LoanStatus                           113937 non-null  object 
 6   ClosedDate                           55089 non-null   object 
 7   BorrowerAPR                          113912 non-null  float64
 8   BorrowerRate                         113937 non-null  float64
 9   LenderYield                          113937 non-null  float64
 10  EstimatedEffectiveYield              84853 non-null   float64
 11  EstimatedLoss

In [5]:
# Columns to keep
col1 = ['ListingKey','ListingNumber','ListingCreationDate','CreditGrade','Term','LoanStatus','CloseDate','BorrowerAPR','BorrowerRate',
    'LenderYield','EstimatedEffectiveYield','EstimatedLoss','EstimatedReturn','ProsperRating (numeric)','ProsperRating (Alpha)',
    'ProsperScore','ListingCategory (numeric)','BorrowerState','Occupation','ExmploymentStatus','ExmploymentStatusDuration']

In [6]:
loanDFsub1 = loanDF.iloc[:,1:20]
loanDFsub1.head(10)

ListingNumber            ListingCreationDate CreditGrade  Term LoanStatus  \
0         193129  2007-08-26 19:09:29.263000000           C    36  Completed   
1        1209647  2014-02-27 08:28:07.900000000         NaN    36    Current   
2          81716  2007-01-05 15:00:47.090000000          HR    36  Completed   
3         658116  2012-10-22 11:02:35.010000000         NaN    36    Current   
4         909464  2013-09-14 18:38:39.097000000         NaN    36    Current   
5        1074836  2013-12-14 08:26:37.093000000         NaN    60    Current   
6         750899  2013-04-12 09:52:56.147000000         NaN    36    Current   
7         768193  2013-05-05 06:49:27.493000000         NaN    36    Current   
8        1023355  2013-12-02 10:43:39.117000000         NaN    36    Current   
9        1023355  2013-12-02 10:43:39.117000000         NaN    36    Current   

            ClosedDate  BorrowerAPR  BorrowerRate  LenderYield  \
0  2009-08-14 00:00:00      0.16516        0.1580       0.1380   
1                  NaN      0.12016        0.0920       0.0820   
2  2009-12-17 00:00:00      0.28269        0.2750       0.2400   
3                  NaN      0.12528        0.0974       0.0874   
4                  NaN      0.24614        0.2085       0.1985   
5                  NaN      0.15425        0.1314       0.1214   
6                  NaN      0.31032        0.2712       0.2612   
7                  NaN      0.23939        0.2019       0.1919   
8                  NaN      0.07620        0.0629       0.0529   
9                  NaN      0.07620        0.0629       0.0529   

   EstimatedEffectiveYield  EstimatedLoss  EstimatedReturn  \
0                      NaN            NaN              NaN   
1                  0.07960         0.0249          0.05470   
2                      NaN            NaN              NaN   
3                  0.08490         0.0249          0.06000   
4                  0.18316         0.0925          0.09066   
5                  0.11567         0.0449          0.07077   
6                  0.23820         0.1275          0.11070   
7                  0.17830         0.0799          0.09840   
8                  0.05221         0.0099          0.04231   
9                  0.05221         0.0099          0.04231   

   ProsperRating (numeric) ProsperRating (Alpha)  ProsperScore  \
0                      NaN                   NaN           NaN   
1                      6.0                     A           7.0   
2                      NaN                   NaN           NaN   
3                      6.0                     A           9.0   
4                      3.0                     D           4.0   
5                      5.0                     B          10.0   
6                      2.0                     E           2.0   
7                      4.0                     C           4.0   
8                      7.0                    AA           9.0   
9                      7.0                    AA          11.0   

   ListingCategory (numeric) BorrowerState      Occupation EmploymentStatus  
0                          0            CO           Other    Self-employed  
1                          2            CO    Professional         Employed  
2                          0            GA           Other    Not available  
3                         16            GA   Skilled Labor         Employed  
4                          2            MN       Executive         Employed  
5                          1            NM    Professional         Employed  
6                          1            KS  Sales - Retail         Employed  
7                          2            CA         Laborer         Employed  
8                          7            IL    Food Service         Employed  
9                          7            IL    Food Service         Employed

In [7]:
loanDFsub2 = loanDF.iloc[:,20:40]
loanDFsub2.head(10)

EmploymentStatusDuration  IsBorrowerHomeowner  CurrentlyInGroup  \
0                       2.0                 True              True   
1                      44.0                False             False   
2                       NaN                False              True   
3                     113.0                 True             False   
4                      44.0                 True             False   
5                      82.0                 True             False   
6                     172.0                False             False   
7                     103.0                False             False   
8                     269.0                 True             False   
9                     269.0                 True             False   

                  GroupKey               DateCreditPulled  \
0                      NaN  2007-08-26 18:41:46.780000000   
1                      NaN            2014-02-27 08:28:14   
2  783C3371218786870A73D20  2007-01-02 14:09:10.060000000   
3                      NaN            2012-10-22 11:02:32   
4                      NaN            2013-09-14 18:38:44   
5                      NaN            2013-12-14 08:26:40   
6                      NaN            2013-04-12 09:52:53   
7                      NaN            2013-05-05 06:49:25   
8                      NaN            2013-12-02 10:43:39   
9                      NaN            2013-12-02 10:43:39   

   CreditScoreRangeLower  CreditScoreRangeUpper FirstRecordedCreditLine  \
0                  640.0                  659.0     2001-10-11 00:00:00   
1                  680.0                  699.0     1996-03-18 00:00:00   
2                  480.0                  499.0     2002-07-27 00:00:00   
3                  800.0                  819.0     1983-02-28 00:00:00   
4                  680.0                  699.0     2004-02-20 00:00:00   
5                  740.0                  759.0     1973-03-01 00:00:00   
6                  680.0                  699.0     2000-09-29 00:00:00   
7                  700.0                  719.0     1999-02-25 00:00:00   
8                  820.0                  839.0     1993-04-01 00:00:00   
9                  820.0                  839.0     1993-04-01 00:00:00   

   CurrentCreditLines  OpenCreditLines  TotalCreditLinespast7years  \
0                 5.0              4.0                        12.0   
1                14.0             14.0                        29.0   
2                 NaN              NaN                         3.0   
3                 5.0              5.0                        29.0   
4                19.0             19.0                        49.0   
5                21.0             17.0                        49.0   
6                10.0              7.0                        20.0   
7                 6.0              6.0                        10.0   
8                17.0             16.0                        32.0   
9                17.0             16.0                        32.0   

   OpenRevolvingAccounts  OpenRevolvingMonthlyPayment  InquiriesLast6Months  \
0                      1                         24.0                   3.0   
1                     13                        389.0                   3.0   
2                      0                          0.0                   0.0   
3                      7                        115.0                   0.0   
4                      6                        220.0                   1.0   
5                     13                       1410.0                   0.0   
6                      6                        214.0                   0.0   
7                      5                        101.0                   3.0   
8                     12                        219.0                   1.0   
9                     12                        219.0                   1.0   

   TotalInquiries  CurrentDelinquencies  AmountDelinquent  \
0             3.0               

In [8]:
loanDFsub3 = loanDF.iloc[:,40:60]
loanDFsub3.head(10)

RevolvingCreditBalance  BankcardUtilization  AvailableBankcardCredit  \
0                     0.0                 0.00                   1500.0   
1                  3989.0                 0.21                  10266.0   
2                     NaN                  NaN                      NaN   
3                  1444.0                 0.04                  30754.0   
4                  6193.0                 0.81                    695.0   
5                 62999.0                 0.39                  86509.0   
6                  5812.0                 0.72                   1929.0   
7                  1260.0                 0.13                   2181.0   
8                  9906.0                 0.11                  77696.0   
9                  9906.0                 0.11                  77696.0   

   TotalTrades  TradesNeverDelinquent (percentage)  TradesOpenedLast6Months  \
0         11.0                                0.81                      0.0   
1         29.0                                1.00                      2.0   
2          NaN                                 NaN                      NaN   
3         26.0                                0.76                      0.0   
4         39.0                                0.95                      2.0   
5         47.0                                1.00                      0.0   
6         16.0                                0.68                      0.0   
7         10.0                                0.80                      0.0   
8         29.0                                1.00                      1.0   
9         29.0                                1.00                      1.0   

   DebtToIncomeRatio     IncomeRange  IncomeVerifiable  StatedMonthlyIncome  \
0               0.17  $25,000-49,999              True          3083.333333   
1               0.18  $50,000-74,999              True          6125.000000   
2               0.06   Not displayed              True          2083.333333   
3               0.15  $25,000-49,999              True          2875.000000   
4               0.26       $100,000+              True          9583.333333   
5               0.36       $100,000+              True          8333.333333   
6               0.27  $25,000-49,999              True          2083.333333   
7               0.24  $25,000-49,999              True          3355.750000   
8               0.25  $25,000-49,999              True          3333.333333   
9               0.25  $25,000-49,999              True          3333.333333   

                   LoanKey  TotalProsperLoans  TotalProsperPaymentsBilled  \
0  E33A3400205839220442E84                NaN                         NaN   
1  9E3B37071505919926B1D82                NaN                         NaN   
2  6954337960046817851BCB2                NaN                         NaN   
3  A0393664465886295619C51                NaN                         NaN   
4  A180369302188889200689E                1.0                        11.0   
5  C3D63702273952547E79520                NaN                         NaN   
6  CE963680102927767790520                NaN                         NaN   
7  0C87368108902149313D53B                NaN                         NaN   
8  02163700809231365A56A1C                NaN                         NaN   
9  02163700809231365A56A1C                NaN                         NaN   

   OnTimeProsperPayments  ProsperPaymentsLessThanOneMonthLate  \
0                    NaN                                  NaN   
1                    NaN                                  NaN   
2                    NaN                                  NaN   
3                    NaN                                  NaN   
4                   11.0                                  0.0   
5                    NaN                                  NaN   
6                    NaN                                  NaN   
7                    NaN                                  NaN   
8   

In [9]:
loanDFsub4 = loanDF.iloc[:,60:82]
loanDFsub4.head(10)

LoanFirstDefaultedCycleNumber  LoanMonthsSinceOrigination  LoanNumber  \
0                            NaN                          78       19141   
1                            NaN                           0      134815   
2                            NaN                          86        6466   
3                            NaN                          16       77296   
4                            NaN                           6      102670   
5                            NaN                           3      123257   
6                            NaN                          11       88353   
7                            NaN                          10       90051   
8                            NaN                           3      121268   
9                            NaN                           3      121268   

   LoanOriginalAmount  LoanOriginationDate LoanOriginationQuarter  \
0                9425  2007-09-12 00:00:00                Q3 2007   
1               10000  2014-03-03 00:00:00                Q1 2014   
2                3001  2007-01-17 00:00:00                Q1 2007   
3               10000  2012-11-01 00:00:00                Q4 2012   
4               15000  2013-09-20 00:00:00                Q3 2013   
5               15000  2013-12-24 00:00:00                Q4 2013   
6                3000  2013-04-18 00:00:00                Q2 2013   
7               10000  2013-05-13 00:00:00                Q2 2013   
8               10000  2013-12-12 00:00:00                Q4 2013   
9               10000  2013-12-12 00:00:00                Q4 2013   

                 MemberKey  MonthlyLoanPayment  LP_CustomerPayments  \
0  1F3E3376408759268057EDA              330.43             11396.14   
1  1D13370546739025387B2F4              318.93                 0.00   
2  5F7033715035555618FA612              123.32              4186.63   
3  9ADE356069835475068C6D2              321.45              5143.20   
4  36CE356043264555721F06C              563.97              2819.85   
5  874A3701157341738DE458F              342.37               679.34   
6  AA4535764146102879D5959              122.67              1226.70   
7  737F347089545035681C074              372.60              3353.40   
8  49A53699682291323D04D66              305.54               611.08   
9  49A53699682291323D04D66              305.54               611.08   

   LP_CustomerPrincipalPayments  ...  LP_ServiceFees  LP_CollectionFees  \
0                       9425.00  ...         -133.18                0.0   
1                          0.00  ...            0.00                0.0   
2                       3001.00  ...          -24.20                0.0   
3                       4091.09  ...         -108.01                0.0   
4                       1563.22  ...          -60.27                0.0   
5                        351.89  ...          -25.33                0.0   
6                        604.25  ...          -22.95                0.0   
7                       1955.89  ...          -69.21                0.0   
8                        505.58  ...          -16.77                0.0   
9                        505.58  ...          -16.77                0.0   

   LP_GrossPrincipalLoss  LP_NetPrincipalLoss  \
0                    0.0                  0.0   
1                    0.0                  0.0   
2                    0.0                  0.0   
3                    0.0                  0.0   
4                    0.0                  0.0   
5                    0.0                  0.0   
6                    0.0                  0.0   
7                    0.0                  0.0   
8                    0.0                  0.0   
9                    0.0                  0.0   

   LP_NonPrincipalRecoverypayments  PercentFunded  Recommendations  \
0                              0.0            1.0                0   
1                              0.0            1.0                0   
2                              0.0            1.

## Interesting variables(columns)

In [10]:
loanDF.LoanStatus.value_counts()

Current                   56576
Completed                 38074
Chargedoff                11992
Defaulted                  5018
Past Due (1-15 days)        806
Past Due (31-60 days)       363
Past Due (61-90 days)       313
Past Due (91-120 days)      304
Past Due (16-30 days)       265
FinalPaymentInProgress      205
Past Due (>120 days)         16
Cancelled                     5
Name: LoanStatus, dtype: int64

In [12]:
#df2=df.query("Courses == 'Spark'")

In [13]:
loans_past_due = loanDF.query("LoanStatus in 'Past due'")
loans_past_due.head(5)

Empty DataFrame
Columns: [ListingKey, ListingNumber, ListingCreationDate, CreditGrade, Term, LoanStatus, ClosedDate, BorrowerAPR, BorrowerRate, LenderYield, EstimatedEffectiveYield, EstimatedLoss, EstimatedReturn, ProsperRating (numeric), ProsperRating (Alpha), ProsperScore, ListingCategory (numeric), BorrowerState, Occupation, EmploymentStatus, EmploymentStatusDuration, IsBorrowerHomeowner, CurrentlyInGroup, GroupKey, DateCreditPulled, CreditScoreRangeLower, CreditScoreRangeUpper, FirstRecordedCreditLine, CurrentCreditLines, OpenCreditLines, TotalCreditLinespast7years, OpenRevolvingAccounts, OpenRevolvingMonthlyPayment, InquiriesLast6Months, TotalInquiries, CurrentDelinquencies, AmountDelinquent, DelinquenciesLast7Years, PublicRecordsLast10Years, PublicRecordsLast12Months, RevolvingCreditBalance, BankcardUtilization, AvailableBankcardCredit, TotalTrades, TradesNeverDelinquent (percentage), TradesOpenedLast6Months, DebtToIncomeRatio, IncomeRange, IncomeVerifiable, StatedMonthlyIncome, LoanKey, TotalProsperLoans, TotalProsperPaymentsBilled, OnTimeProsperPayments, ProsperPaymentsLessThanOneMonthLate, ProsperPaymentsOneMonthPlusLate, ProsperPrincipalBorrowed, ProsperPrincipalOutstanding, ScorexChangeAtTimeOfListing, LoanCurrentDaysDelinquent, LoanFirstDefaultedCycleNumber, LoanMonthsSinceOrigination, LoanNumber, LoanOriginalAmount, LoanOriginationDate, LoanOriginationQuarter, MemberKey, MonthlyLoanPayment, LP_CustomerPayments, LP_CustomerPrincipalPayments, LP_InterestandFees, LP_ServiceFees, LP_CollectionFees, LP_GrossPrincipalLoss, LP_NetPrincipalLoss, LP_NonPrincipalRecoverypayments, PercentFunded, Recommendations, InvestmentFromFriendsCount, InvestmentFromFriendsAmount, Investors]
Index: []

[0 rows x 81 columns]

In [14]:
past3160 = loanDF.query("LoanStatus == 'Past Due (31-60 days)'")

In [15]:
past6190 = loanDF.query("LoanStatus == 'Past Due (61-90 days)'")

In [16]:
past91120 = loanDF.loc[loanDF.LoanStatus == 'Past Due (91-120 days)']

In [17]:
past120 = loanDF.query("LoanStatus == 'Past Due (>120 days)'")

In [18]:
defaulted = loanDF.query("LoanStatus == 'Defaulted'")

In [19]:
past3160.head(3)

ListingKey  ListingNumber            ListingCreationDate  \
647   3CAF3580096380178D5C95F         790025  2013-05-28 11:11:58.420000000   
1055  020E3526008577032F83C13         528390  2011-09-20 13:50:34.543000000   
2068  4C02356141219971003613B         661956  2012-10-29 10:53:57.153000000   

     CreditGrade  Term             LoanStatus ClosedDate  BorrowerAPR  \
647          NaN    60  Past Due (31-60 days)        NaN      0.24282   
1055         NaN    36  Past Due (31-60 days)        NaN      0.13413   
2068         NaN    36  Past Due (31-60 days)        NaN      0.35797   

      BorrowerRate  LenderYield  ...  LP_ServiceFees  LP_CollectionFees  \
647         0.2179       0.2079  ...          -24.96            -187.00   
1055        0.1129       0.1029  ...         -196.73            -247.89   
2068        0.3177       0.3077  ...          -45.51             -84.34   

      LP_GrossPrincipalLoss  LP_NetPrincipalLoss  \
647                     0.0                  0.0   
1055                    0.0                  0.0   
2068                    0.0                  0.0   

     LP_NonPrincipalRecoverypayments  PercentFunded  Recommendations  \
647                              0.0            1.0                0   
1055                             0.0            1.0                1   
2068                             0.0            1.0                0   

     InvestmentFromFriendsCount InvestmentFromFriendsAmount Investors  
647                           0                         0.0         1  
1055                          0                         0.0       188  
2068                          0                         0.0        54  

[3 rows x 81 columns]

In [21]:
all_past = past3160.append([past6190, past91120, past120, defaulted])

In [23]:
all_past.sample(5)

ListingKey  ListingNumber            ListingCreationDate  \
105824  CF03339833544726990C6FE         196677  2007-09-03 14:30:44.137000000   
17261   91C03480222573334B045B0         451127  2010-03-19 12:21:01.723000000   
57058   50833400420645883FE40E3         206069  2007-09-24 11:20:40.210000000   
60768   0BE23365696237141081234           6797  2006-04-20 18:19:40.653000000   
40955   147D3522419450411E98291         519465  2011-08-02 10:47:29.493000000   

       CreditGrade  Term LoanStatus           ClosedDate  BorrowerAPR  \
105824           D    36  Defaulted  2008-12-18 00:00:00      0.22231   
17261          NaN    36  Defaulted  2012-08-27 00:00:00      0.19346   
57058           AA    36  Defaulted  2009-05-03 00:00:00      0.10813   
60768            A    36  Defaulted  2007-03-02 00:00:00      0.10692   
40955          NaN    36  Defaulted  2012-12-05 00:00:00      0.27467   

        BorrowerRate  LenderYield  ...  LP_ServiceFees  LP_CollectionFees  \
105824        0.2149       0.2049  ...          -60.61               0.00   
17261         0.1600       0.1500  ...          -15.20               0.00   
57058         0.1012       0.0962  ...          -73.66               0.00   
60768         0.1000       0.0950  ...           -4.98               0.00   
40955         0.2399       0.2299  ...          -32.53             -22.54   

        LP_GrossPrincipalLoss  LP_NetPrincipalLoss  \
105824                6298.36              6298.37   
17261                    0.00                 0.00   
57058                 9703.13              9703.13   
60768                 2196.41              2196.42   
40955                   20.83                 0.00   

       LP_NonPrincipalRecoverypayments  PercentFunded  Recommendations  \
105824                            0.00            1.0                0   
17261                             0.00            1.0                0   
57058                             0.00            1.0                0   
60768                             0.00            1.0                0   
40955                            63.03            1.0                0   

       InvestmentFromFriendsCount InvestmentFromFriendsAmount Investors  
105824                          1                        50.0       156  
17261                           0                         0.0       103  
57058                           0                         0.0       349  
60768                           0                         0.0        59  
40955                           0                         0.0        15  

[5 rows x 81 columns]

In [ ]:
col_names=[]
for column_name in loanDF.columns:
    print(f'{column_name}')
    if "Past Due" in column_name:
        col_names.append(column_name)

print(col_names)

**LoanStatus** could be interesting to investigate further. For example, let's look at all "Past Due..." & look at various vars from there. **IncomeRange**, **CreditGrade**, **Occupation**

In [ ]:
loanDF.CreditGrade.value_counts()

In [ ]:
loanDF.IncomeRange.value_counts()

**IncomeRange** -- another interesting variable to compare with others. I'm particularly interested in the $100,000+ bracket.

In [ ]:
# explore breakdown of Monthly Incomes
pd.set_option('display.float_format', str)
loanDF['StatedMonthlyIncome'] = loanDF['StatedMonthlyIncome'].astype(np.int64)
loanDF['StatedMonthlyIncome'].describe()

In [ ]:
loanDF.Occupation.value_counts().head(30)

In [ ]:
occupation_income = loanDF.groupby('Occupation')['StatedMonthlyIncome'].mean().sort_values(ascending=False)
occupation_income.head(15)

In [ ]:
occupation_income[20:40]

In [ ]:
occupation_income[40:53]

In [ ]:
occupation_income.tail(15)

In [ ]:
loanDFsub1.head()

In [ ]:
loanDFsub2.head()

In [ ]:
loanDFsub3.head()

In [ ]:
loanDF.describe()

### What characteristics correlate with lower interest rate for loans?
### Correlation does NOT equate to causality

In [ ]:
pd.__version__

In [ ]:
# Show mean BorrowerRate by State
BorrowerRatebyState = loanDF.groupby('BorrowerState')['BorrowerRate'].mean().sort_values()
print(f"Top 5 states with lowest average interest rate: \n {BorrowerRatebyState.head(5)}")
print(f"Top 5 states with **highest** interest rate: \n{BorrowerRatebyState.tail(5)}")

In [ ]:
# Show mean Debt-to-Income-Ratio by State
DebtRatiobyState = loanDF.groupby('BorrowerState')['DebtToIncomeRatio'].mean().sort_values()
print(f"Top 5 states with lowest debt to income ratio: \n {DebtRatiobyState.head(5)}")
print(f"Top 5 states with **highest** deb to income ratio: \n{DebtRatiobyState.tail(5)}")

In [ ]:
# How many loans are there of each ListingCategory
loanDF['ListingCategory (numeric)'].value_counts().sort_values()

#### 1-Debt Consolidation @ 58308, 0-Not Avail., 7-other, 2-Home Improvement @ 7433, 3-Business @ 7189, 6-Auto @ 2572 

The category of the listing that the borrower selected when posting their listing: <br/><br/>
0 - Not Available, 1 - Debt Consolidation, 2 - Home Improvement, <br/>
3 - Business, 4 - Personal Loan, 5 - Student Use, <br/>
6 - Auto, 7- Other, 8 - Baby&Adoption, 9 - Boat, <br/>
10 - Cosmetic Procedure, 11 - Engagement Ring, <br/>
12 - Green Loans, 13 - Household Expenses, 14 - Large Purchases, <br/>
15 - Medical/Dental, 16 - Motorcycle, 17 - RV, <br/>
18 - Taxes, 19 - Vacation, 20 - Wedding Loans <br/>

## Categorical columns converted to ordered category

In [ ]:
# convert CreditGrade, ProsperRating (Alpha), IncomeRange into ordered catebgory
#ordinal_var_dict = {'CreditGrade': ['

### What is the structure of your dataset?

> The Prosper Loan Data set is 113937 records with 81 variable/columns. 

### What is/are the main feature(s) of interest in your dataset?

> Greatest interest so far: loan amount, interest rate, term. The items that make the loan most attractive and least attractive on the opposite side of the spectrum  . Any other trends that are peculiar, etc.

### What features in the dataset do you think will help support your investigation into your feature(s) of interest?

> Your answer here!

## Univariate Exploration

> In this section, investigate distributions of individual variables. If
you see unusual points or outliers, take a deeper look to clean things up
and prepare yourself to look at relationships between variables.

In [ ]:
sb.distplot(loanDF['BorrowerAPR']);

In [ ]:
borrAPR = loanDF.query('BorrowerAPR > 0.35 and BorrowerAPR < 0.37')
borrAPR.shape

In [ ]:
sb.distplot(borrAPR['BorrowerRate']);

In [ ]:
# Bin resizing, to transform the x-axis    
#bin_edges = np.arange(0, sqrt_trans(pokemon['weight'].max())+1, 1)
plt.figure(figsize = [15,10])
# Plot the scaled data
plt.hist(loanDF['BorrowerRate'])

# Identify the tick-locations
#tick_locs = np.arange(0, 21, 1)

# Apply x-ticks
#plt.xticks(tick_locs, [str(x) for x in np.arange(0,21,1)]);

### What is going on with the BorrowerAPR/Rate at 31-33% & 36-37%, well over 5000 records? Let's investigate additional info about these records

In [ ]:
sb.distplot(loanDF['BorrowerRate']);

In [ ]:
BorrRate30 = loanDF.query("BorrowerRate <= 0.33 and BorrowerRate >= 0.30")
BorrRate30.shape

## Loan Terms = some 12 month, majority are 36 mnth loans, but healthy percentage at 60 mnth (5 year loan)

In [ ]:
sb.distplot(loanDF['Term']);

In [ ]:
# Use Seaborn's CountPlot as an alternative to compare with plt.hist & seaborn's distplot

base_color = sb.color_palette()[2]
sb.countplot(data = loanDF, x = 'ListingCategory (numeric)', color = base_color);

In [ ]:
# Plot the scaled data
plt.hist(loanDF['LoanOriginalAmount']);

# Identify the tick-locations
#tick_locs = np.arange(0, 21, 1)

# Apply x-ticks
#plt.xticks(tick_locs, [str(x) for x in np.arange(0,21,1)]);

In [ ]:
sb.distplot(loanDF['LoanOriginalAmount']);

In [ ]:
base_color = sb.color_palette()[2]
sb.countplot(data = loanDF, x = 'LoanOriginalAmount', bins=10, color = base_color);

In [ ]:
help(sb.distplot)

> Make sure that, after every plot or related series of plots, that you
include a Markdown cell with comments about what you observed, and what
you plan on investigating next.

### Discuss the distribution(s) of your variable(s) of interest. Were there any unusual points? Did you need to perform any transformations?

> Your answer here!

### Of the features you investigated, were there any unusual distributions? Did you perform any operations on the data to tidy, adjust, or change the form of the data? If so, why did you do this?

> Your answer here!

## Bivariate Exploration

> In this section, investigate relationships between pairs of variables in your
data. Make sure the variables that you cover here have been introduced in some
fashion in the previous section (univariate exploration).

### Talk about some of the relationships you observed in this part of the investigation. How did the feature(s) of interest vary with other features in the dataset?

> Your answer here!

### Did you observe any interesting relationships between the other features (not the main feature(s) of interest)?

> Your answer here!

## Multivariate Exploration

> Create plots of three or more variables to investigate your data even
further. Make sure that your investigations are justified, and follow from
your work in the previous sections.

### Talk about some of the relationships you observed in this part of the investigation. Were there features that strengthened each other in terms of looking at your feature(s) of interest?

> Your answer here!

### Were there any interesting or surprising interactions between features?

> Your answer here!

> At the end of your report, make sure that you export the notebook as an
html file from the `File > Download as... > HTML` menu. Make sure you keep
track of where the exported file goes, so you can put it in the same folder
as this notebook for project submission. Also, make sure you remove all of
the quote-formatted guide notes like this one before you finish your report!